Sources/PointFree/Subscribe.swift

```diff
@@ -6,19 +6,35 @@ import Prelude
 import Tuple
 
 public struct SubscribeData: Codable {
-  public let pricing: Pricing
-  public let token: Stripe.Token.Id
-  public let vatNumber: String
+  public typealias Coupon = Tagged<SubscribeData, String>
+
+  public private(set) var coupon: Coupon?
+  public private(set) var pricing: Pricing
+  public private(set) var token: Stripe.Token.Id
+  public private(set) var vatNumber: String
 }
 
 let subscribeMiddleware =
   filterMap(
     require1 >>> pure,
-    or: redirect(to: .pricing(nil, expand: nil), headersMiddleware: flash(.error, "Error creating subscription!"))
+    or: redirect(
+      to: .pricing(nil, expand: nil),
+      headersMiddleware: flash(.error, "Error creating subscription!")
+    )
     )
     <<< filter(
       get1 >>> ^\.pricing >>> validateQuantity,
-      or: redirect(to: .pricing(nil, expand: nil), headersMiddleware: flash(.error, "An invalid subscription quantity was used."))
+      or: redirect(
+        to: .pricing(nil, expand: nil),
+        headersMiddleware: flash(.error, "An invalid subscription quantity was used.")
+      )
+    )
+    <<< filter(
+      get1 >>> validateCoupon(forSubscribeData:),
+      or: redirect(
+        to: .pricing(nil, expand: nil),
+        headersMiddleware: flash(.error, "Coupons can only be used on individual subscription plans.")
```

Comment:
right now I'm limiting discounts to only individual subscriptions. you think that makes sense?

Reply:
Yup! Though we may wanna reuse the same mechanism for team discounts in the future, we can change it then.